src/Parser/ShadowPriest/Modules/Features/AlwaysBeCasting.js

```diff
@@ -0,0 +1,139 @@
+
+        this._highestLingeringStack = this._highestVoidformStack;
+        this.applyHasteGain(this._highestLingeringStack * 0.01);
+        
+        this._highestVoidformStack = 0;
+        
+        debug && console.log(`ABC: Current haste: ${this.currentHaste} (gained ${0.01 * this._highestLingeringStack} from LINGERING_INSANITY)`);
+        return;
+    }
+
+    if (spellId === SPELLS.VOID_TORRENT.id) {
+        return;
+    }
+    
+    super.on_toPlayer_applybuff(event);
+  }
+
+
+  on_toPlayer_applybuffstack(event) {
+    const spellId = event.ability.guid;
+    if (spellId === SPELLS.VOIDFORM_BUFF.id) {
+        this.applyHasteLoss(this._highestVoidformStack * 0.01);
+        this._highestVoidformStack = event.stack;
+        this.applyHasteGain(this._highestVoidformStack * 0.01);
+
+        debug && console.log(`ABC: Current haste: ${this.currentHaste} (gained ${0.01 * this._highestVoidformStack} from VOIDFORM_BUFF)`);
+    }
+    
+  }
+
+  on_toPlayer_removebuffstack(event) {
+    const spellId = event.ability.guid;
+    if (spellId === SPELLS.LINGERING_INSANITY.id) {
+        this.applyHasteLoss(this._highestLingeringStack * 0.01);
+        this._highestLingeringStack = event.stack;
+        this.applyHasteGain(this._highestLingeringStack * 0.01);
+
+        debug && console.log(`ABC: Current haste: ${this.currentHaste} (lost ${0.02} from LINGERING_INSANITY)`);
+    }
+    
+  }
+
+  on_toPlayer_removebuff(event) {
+    const spellId = event.ability.guid;
+    if(
+        spellId === SPELLS.MIND_FLAY.id ||
+        spellId === SPELLS.DISPERSION.id ||
+        spellId === SPELLS.VOID_TORRENT.id
+    ){
+        this.lastCastFinishedTimestamp = event.timestamp;
+        return;
+    }
+
+
+    super.on_toPlayer_removebuff(event);
+  }
+
+  on_finished() {
+    const fightDuration = this.owner.fight.end_time - this.owner.fight.start_time;
+    debug && console.log('totalTimeWasted:', this.totalTimeWasted, 'totalTime:', fightDuration, (this.totalTimeWasted / fightDuration));
+    debug && console.log('totalHealingTimeWasted:', this.totalHealingTimeWasted, 'totalTime:', fightDuration, (this.totalHealingTimeWasted / fightDuration));
+
+    // override default as it never does the voidform/lingering insanity haste changes
+  }
+
+  suggestions(when) {
+    const deadTimePercentage = this.totalTimeWasted / this.owner.fightDuration;
+
+    when(deadTimePercentage).isGreaterThan(0.2)
+      .addSuggestion((suggest, actual, recommended) => {
+        return suggest(<span>Your dead GCD time can be improved. Try to Always Be Casting (ABC), try to reduce the delay between casting spells. Even if you have to move, try casting something instant - maybe refresh your dots or replenish your mana with <SpellLink id={SPELLS.LIFE_TAP.id}/></span>)
+          .icon('spell_mage_altertime')
+          .actual(`${formatPercentage(deadTimePercentage)}% dead GCD time`)
```

Comment:
Using the `actual` parameter here instead makes it so you don't have to keep the two values in sync when you change the `when` value.